cauldron/sql.py

```diff
@@ -186,7 +200,10 @@ def get_pool(cls) -> Pool:
         if not cls._pool:
             with (yield from cls._pool_pending):
                 if not cls._pool:
-                    cls._pool = yield from create_pool(**cls._connection_params)
+                    if is_replica:
+                        cls._pool = yield from create_pool(**cls._replica_connection_params)
```

Comment:
two pool should be created one should be cls._pool and other should be cls._replica_pool otherwise concurrent request scope will be changed on the fly. Change this method